libsolidity/parsing/Parser.cpp

```diff
@@ -111,7 +111,21 @@ ASTPointer<SourceUnit> Parser::parse(shared_ptr<Scanner> const& _scanner)
 				nodes.push_back(parseFunctionDefinition(true));
 				break;
 			default:
-				fatalParserError(7858_error, "Expected pragma, import directive or contract/interface/library/struct/enum/function definition.");
+				if (
+					m_scanner->currentToken() == Token::Identifier ||
+					m_scanner->currentToken() == Token::Mapping ||
```

Comment:
Constant mapping?

Reply:
This is just the generic code for variable declaration, I'll probably pull it out.